src/pages/Products/Products.tsx

```diff
@@ -69,19 +76,18 @@ export default function Products() {
         </div>
         <div className={cls.products_info}>
           <h2>Total Product</h2>
-          <div className={cls._count}>{totalProducts.length}</div>
+          <div className={cls.count}>{totalProducts.length}</div>
         </div>
       </div>
       <InfiniteScroll
+        className={cls.infinite_scroll}
         dataLength={products.length}
-        next={() => setPage(page + 10)}
-        hasMore={products.length < 200}
-        loader={<Loader size={LoaderSize.l} />}
-        endMessage={
-          <p style={{ textAlign: "center" }}>
-            <b>You have seen it all</b>
-          </p>
+        next={() => setPage((prevPage) => prevPage + 10)}
+        hasMore={
+          products.length === 0 || totalProducts.length !== products.length
```

Comment:
если вдруг в промежуток времени между тем, как ты получил totalProducts и получил последнюю пачку товаров, добавится ещё товаров, то у тебя `totalProducts.length !== products.length` будет false и станет показываться лоадер и, возможно, идти лишние запросы на бэк. безопаснее написать:
```suggestion
          products.length === 0 || totalProducts.length <= products.length
```